### ISSUE TITLE
Error: edition not found

### ISSUE THREAD

**Issue by spyroot** [vmware-tanzu/tanzu-cli#717]
Hi Folks,

It doesn't matter what you do; the tool always reports an error that the edition has not been found.

```
tanzu management-cluster create
Error: edition not found
```

it is not clear where that key should be in config-ng.yaml ( by default generated) or config.yaml (generated)

in all cases, I tried to add key edition: tag in different spots in the same story.

Docs said it should be set to tkg by default.

```
tanzu config set edition tkg
[x] : unable to parse config path parameter into parts [edition]  (was expecting 'features.<plugin>.<feature>' or 'env.<env_variable>')
```

```
tanzu config set cli.edition tkg
[x] : unsupported config path parameter [cli] (was expecting 'features.<plugin>.<feature>' or 'env.<env_variable>')
```

The only one it takes env.edition tkg

But none of these options run.
Note I did not indicate the rest --ui, etc. since the error was thrown before, so I suspect it parsed that read config.yaml before the rest of the args.

```
tanzu version
version: v1.2.0
buildDate: 2024-02-07
sha: f3abe62e
arch: amd64

```

Trie don mac same story

**Comment by marckhouzam** [comment]
Hi @spyroot and sorry about the issue you are facing.

CLI version `v1.2.0` is not supported with TKG.  You need to revert to using `v1.1.0`

**Comment by marckhouzam** [comment]
You can confirm what is compatible here: https://interopmatrix.vmware.com/Interoperability?col=1772,&row=0,

**Comment by spyroot** [comment]
Yes, but why does it expect a key edition in the first place?   In essence, if you download from the latest from VMware, the first command in docs says tanzu create mc -ui but it will fail with the default config.yaml tool generated itself.  
If it does not support that, that is okay, but then it should include a key in the configuration.yaml it expects

In essence what edition supported as key value and how it should look like ? 
Because I check unit test it doesn't look like it same format.

**Comment by marckhouzam** [comment]
The CLI itself no longer writes the edition in the config file but when your run `tanzu mc` you are running the management-cluster plugin from TKG which has not been updated to no longer look for the edition. 

If you want to see what the edition entry should look like in the config you can run an older CLI once (e.g., simply `tanzu.1.1.0 version`) and it will automatically add the key.

**Comment by MAHDTech** [comment]
Thanks for this, hit the same issue 👍🏻
